### PR TITLE
Add coveralls gem for test coverage metrics

### PR DESCRIPTION
**Pull request by koggs**
### What we did:
- Set up Coveralls for test coverage
- Configure Coveralls for Rspec and Cucumber

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -1,58 +1,15 @@
-# IMPORTANT: This file is generated by cucumber-rails - edit at your own peril.
-# It is recommended to regenerate this file in the future when you upgrade to a
-# newer version of cucumber-rails. Consider adding your own code to a new file
-# instead of editing this one. Cucumber will automatically load all features/**/*.rb
-# files.
+require 'coveralls'
+Coveralls.wear_merged!
```
Comment (alghazu) [inline features/support/env.rb]: I think should be:
`Coveralls.wear_merged!('rails')`

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,96 +1,15 @@
...
-# light-weight as possible. Requiring heavyweight dependencies from this file
-# will add to the boot time of your test suite on EVERY test run, even for an
-# individual file that may not need all of that loaded. Instead, consider making
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+require 'coveralls'
+Coveralls.wear_merged!
```
Comment (alghazu) [inline spec/spec_helper.rb]: I think should be:
`Coveralls.wear_merged!('rails')`

**Review by alghazu — CHANGES_REQUESTED** [review]

File: spec/spec_helper.rb (hunk 2 above)
Comment (koggs) [reply spec/spec_helper.rb]: I fixed it

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,96 +1,15 @@
...
-# will add to the boot time of your test suite on EVERY test run, even for an
-# individual file that may not need all of that loaded. Instead, consider making
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+require 'coveralls'
+Coveralls.wear_merged!('rails')
+
```
Comment (faraznaeem) [inline spec/spec_helper.rb]: This code should be in the rails helper

File: coverage/.resultset.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (faraznaeem) [inline coverage/.resultset.json]: This file should be removed from version control

File: coverage/.last_run.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "result": {
+    "covered_percent": 0.0
+  }
+}
```
Comment (faraznaeem) [inline coverage/.last_run.json]: this file should be removed from version control

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: coverage/.last_run.json (hunk 5 above)
Comment (koggs) [reply coverage/.last_run.json]: Comments have been addressed

**Comment by alghazu** [comment]
Done

**Review by faraznaeem — APPROVED** [review]

**Review by alghazu — APPROVED** [review]

**Review by GeorgeJLewis — APPROVED** [review]

**Comment by GeorgeJLewis** [comment]
LGTM

**Pull request merged**